modules/effects/src/effects_resolver.ts

```diff
@@ -13,7 +13,8 @@ export function mergeEffects(
   globalErrorHandler: ErrorHandler,
   effectsErrorHandler: EffectsErrorHandler
 ): Observable<EffectNotification> {
-  const sourceName = getSourceForInstance(sourceInstance).constructor.name;
+  const source = getSourceForInstance(sourceInstance);
+  const sourceName = source ? source.constructor.name : 'Object';
```

Comment:
Unrelated, but seeing this made me wonder how `sourceName` is used throughout the whole flow.
I ended up that it's only used to throw an error when an effect doesn't dispatch an action (an object with `type` as property).
Do we want to change the message for functional effects?
It currently throws the following message:

```js
ERROR Error: Effect "Object.logDispatchedActions()" dispatched an invalid action: {}
``` 

We can prettify this a little bit to

```js
//           👇 does this add value?
//                                👇 remove `Object.`
ERROR Error: (Functional) Effect "logDispatchedActions()" dispatched an invalid action: {}
```

Reply:
Good point!